v2/api/network/customizations/private_endpoints_extensions.go

```diff
@@ -60,3 +67,99 @@ func (extension *PrivateEndpointExtension) PostReconcileCheck(
 
 	return extensions.PostReconcileCheckResultSuccess(), nil
 }
+
+func (extension *PrivateEndpointExtension) ExportKubernetesResources(
+	ctx context.Context,
+	obj genruntime.MetaObject,
+	armClient *genericarmclient.GenericClient,
+	log logr.Logger,
+) ([]client.Object, error) {
+	// This has to be the current hub storage version. It will need to be updated
+	// if the hub storage version changes.
+	endpoint, ok := obj.(*network.PrivateEndpoint)
+	if !ok {
+		return nil, errors.Errorf("cannot run on unknown resource type %T, expected *network.PrivateEndpoint", obj)
+	}
+
+	// Type assert that we are the hub type. This will fail to compile if
+	// the hub type has been changed but this extension has not
+	var _ conversion.Hub = endpoint
+
+	hasIpConfiguration := configMapSpecified(endpoint)
+	if !hasIpConfiguration {
+		log.V(Debug).Info("no configmap retrieval to perform as operatorSpec is empty")
+		return nil, nil
+	}
+
+	if endpoint.Status.NetworkInterfaces == nil || len(endpoint.Status.NetworkInterfaces) == 0 {
+		log.V(Debug).Info("no configmap retrieval to perform as there is no NetworkInterfaces attached")
+		return nil, nil
+	}
+
+	nicId, err := arm.ParseResourceID(*endpoint.Status.NetworkInterfaces[0].Id)
+	if err != nil {
+		return nil, err
+	}
+
+	// The default primary ip configuration for PrivateEndpoint is on NetworkInterfaceController. Hence, we fetch it from there.
+	var interfacesClient *armnetwork.InterfacesClient
+	interfacesClient, err = armnetwork.NewInterfacesClient(nicId.SubscriptionID, armClient.Creds(), armClient.ClientOptions())
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create new PrivateEndpointsClient")
+	}
+
+	var resp armnetwork.InterfacesClientGetResponse
+	resp, err = interfacesClient.Get(ctx, nicId.ResourceGroupName, nicId.Name, nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed getting PrivateEndpoint")
+	}
+
+	configsByName := configByName(resp)
+	configs, err := configMapToWrite(endpoint, configsByName)
+	if err != nil {
+		return nil, err
+	}
+
+	return configmaps.SliceToClientObjectSlice(configs), nil
+}
+
+func configByName(resp armnetwork.InterfacesClientGetResponse) map[string]string {
+	result := make(map[string]string)
+
+	if resp.Properties != nil && resp.Properties.IPConfigurations != nil && len(resp.Properties.IPConfigurations) > 0 {
+		result["PrimaryNicPrivateIPAddress"] = *resp.Properties.IPConfigurations[0].Properties.PrivateIPAddress
+	}
+
+	return result
+}
+
+func configMapToWrite(obj *network.PrivateEndpoint, configs map[string]string) ([]*v1.ConfigMap, error) {
+	operatorSpecConfigs := obj.Spec.OperatorSpec.ConfigMaps
+	if operatorSpecConfigs == nil {
+		return nil, errors.Errorf("unexpected nil operatorspec")
+	}
+
+	collector := configmaps.NewCollector(obj.Namespace)
+
+	primaryNicPrivateIPAddress, ok := configs["PrimaryNicPrivateIPAddress"]
```

Comment:
```suggestion
	primaryNICPrivateIPAddress, ok := configs["PrimaryNicPrivateIPAddress"]
```

Reply:
(but as suggested above maybe do away with the map entirely here)

Reply:
I'd rather prefer `primaryNicPrivateIPAddress` here to keep it consistent with the operatorSpec property name?